src/app/plugins/alt.plugin.ts

```diff
@@ -0,0 +1,175 @@
+    const [subCommand, oldUser, newUser] = args.join(' ').split('\n');
+
+    // Try to resolve firstUser
+    const oldMember = await Moderation.Helpers.resolveUser(
+      this.container.guildService.get(),
+      oldUser
+    );
+
+    if (!oldMember) {
+      await message.reply('First user not found');
+      return;
+    }
+
+    // Add command given
+    if (subCommand === 'add') {
+      // Try to resolve the second user
+      const newMember = await Moderation.Helpers.resolveUser(
+        this.container.guildService.get(),
+        newUser
+      );
+      if (!newMember) {
+        await message.reply('Second user not found');
+        return;
+      }
+
+      // Link the 2 IDs in the DB
+      const tracker = await this._handleAdd(oldMember.id, newMember.id);
+      const { baseID, knownIDs } = tracker;
+
+      await message.reply({
+        embeds: [
+          (await this._createAssociationEmbed(baseID, knownIDs))
+            .setTitle('Association Created')
+            .setDescription(
+              `An association between \`${oldUser}\` and \`${newUser}\` has been created.`
+            ),
+        ],
+      });
+
+      return;
+    }
+
+    // List command given
+    if (subCommand === 'list') {
+      const knownIDs = await this.container.modService.getAllKnownAltIDs(
+        this.container.guildService.get(),
+        oldUser
+      );
+      await message.reply({
+        embeds: [
+          (await this._createAssociationEmbed(oldMember.id, knownIDs))
+            .setTitle(`All known IDs for \`${oldUser}\``)
+            .setDescription(
+              knownIDs.length > 1
+                ? `User has \`${knownIDs.length - 1}\` known alt(s)`
+                : 'User has no known alts'
+            ),
+        ],
+      });
+
+      return;
+    }
+
+    // This shouldn't happen, but leaving here for future subcommands
+    await message.reply('Unknown subcommand');
+  }
+
+  private async _handleAdd(oldID: string, newID: string) {
+    let match = await AltTrackerModel.findOne({
+      guildID: this.container.guildService.get().id,
+      $or: [{ baseID: oldID }, { baseID: newID }],
+    });
+
+    // If this is the first alt, make a new entry
+    if (!match) {
+      match = await this._createEntry(oldID);
+    }
+
+    // Make sure both IDs are in the knownIDs
+    // This also prevents duplicate entries
+    if (!match.knownIDs.includes(oldID)) {
+      match.knownIDs.push(oldID);
+    }
+
+    if (!match.knownIDs.includes(newID)) {
+      match.knownIDs.push(newID);
+    }
+
+    return match.save();
+  }
+
+  private async _createEntry(oldID: string) {
+    return AltTrackerModel.create({
+      guildID: this.container.guildService.get().id,
+      baseID: oldID,
+      knownIDs: [],
+    });
+  }
+
+  private async _createAssociationEmbed(baseID: string, knownIDs: string[]) {
+    const members = Array.from((await this.container.guildService.get().members.fetch()).values());
```

Comment:
I did this to make the function signature for `_tryToConvertToGuildMember` nicer.

But it would be O(1) if I just use the collection that `members.fetch()` gives right?
Assuming that's what ur talking about

Reply:
```ts
const members = await this.container.guildService.get().members.fetch();
...

private _tryToConvertToGuildMember(members: GuildMember[], id: string): GuildMember | string {
    const member = members.get(id);
    ...
```
?